numpy/lib/_function_base_impl.py

```diff
@@ -538,9 +548,19 @@ def average(a, axis=None, weights=None, returned=False, *,
                     "Length of weights not compatible with specified axis.")
 
             # setup wgt to broadcast along axis
-            wgt = np.broadcast_to(wgt, (a.ndim-1)*(1,) + wgt.shape)
+            if ignore_nan:
+                wgt = np.broadcast_to(wgt, a.shape)
```

Comment:
I don't fully understand this part. Can you add a comment? Why when ignoring nans we can just BC `wgt` to `a`'s shape?

Reply:
My intention here was to fully broadcast wgt to match a's that way if:

```python 
a = [[1, 2, 3, np.nan], [1, 2, 3, 4]]
wgt = [1, 1, 1, 1]
wgt_broadcasted = [[1, 1, 1, 1], [1, 1, 1, 1]]
wgt_nan_removed = [[1, 1, 1, 0], [1, 1, 1, 1]]
ans = [2, 2.5]
```

Without fully broadcasting from the start I believe I would have to set `wgt=[1, 1, 1, 0]` for all the axes. I could be misunderstanding what's going on here though. 

Seems like either way I'll be closing this.